app/src/main/java/org/sil/gatherwords/MainActivity.java

```diff
@@ -165,7 +175,84 @@ public void onClick(View v) {
                 }
             });
 
+            ImageButton deleteButton = convertView.findViewById(R.id.session_list_button_delete);
+            deleteButton.setOnClickListener(new View.OnClickListener() {
+                public void onClick(View v) {
+                    Context context = inflater.getContext();
+                    new DeleteSessionFromDB(context).execute(session);
+
+                }
+            });
+
             return convertView;
         }
     }
+
+    private static class DeleteSessionFromDB extends AsyncTask<Session, Void, List<Session>> {
+        private SessionDao sDAO;
+        private WeakReference<MainActivity> mainActivityRef;
+
+        DeleteSessionFromDB(Context context) {
+            sDAO = AppDatabase.get(context).sessionDao();
+            mainActivityRef = new WeakReference<>((MainActivity) context);
+        }
+
+        @Override
+        protected List<Session> doInBackground(Session... sessions) {
+            sessions[0].deletedAt = new Date();
```

Comment:
I would recommend:
```java
for (Session session : sessions) {
    session.deletedAt = new Date();
}
```
This avoids the assumption that `sessions` will always contain exactly 1 element.